gatewayconfig/bluetooth/characteristics/add_gateway_characteristic.py

```diff
@@ -52,32 +77,27 @@ def WriteValue(self, value, options):
 
             addGatewayDetails = add_gateway_pb2.add_gateway_v1()
             addGatewayDetails.ParseFromString(bytes(value))
-            miner_bus = dbus.SessionBus()
-
-            logger.debug("Loading dbus com.helium.Miner")
-            miner_object = miner_bus.get_object('com.helium.Miner', '/')
-            sleep(DBUS_LOAD_SLEEP_SECONDS)
-            miner_interface = dbus.Interface(miner_object, 'com.helium.Miner')
-            sleep(DBUS_LOAD_SLEEP_SECONDS)
-
-            logger.debug("Parsing onboarding values")
-            owner = addGatewayDetails.owner
-            fee = addGatewayDetails.fee
-            amount = addGatewayDetails.amount
-            payer = addGatewayDetails.payer
-
-            logger.debug("Registering owner %s, fee %s, amount %s, payer %s" % (owner, fee, amount, payer))
-            # Calls https://github.com/helium/miner/blob/e55437beac4b46d15cbd079c9c8df045ffc0bf49/src/miner_ebus.erl#L50
-            addMinerRequest = miner_interface.AddGateway(owner, fee, amount, payer)
-            logger.debug("Adding Response")
-            self.notifyValue = addMinerRequest
+
+            logger.debug(f"add gateway owner {addGatewayDetails.owner}, fee {addGatewayDetails.fee} "
+                         f"amount {addGatewayDetails.amount}, payer {addGatewayDetails.payer}")
+            self.add_gateway_details = addGatewayDetails
+            # some thought was given to whether to start the grpc call straight away in a thread/process
+            # and read value from it in readValue. But write and read will come over bluetooth almost back
+            # back. Don't see much value in extra complexity.
         except Exception:
             logger.exception("Unable to register gateway for unknown reason")
 
     def ReadValue(self, options):
         logger.debug('Read Add Gateway')
+        value = self.create_add_gateway_txn()
+        # logger.debug(f"value from grpc call: {value}")
+        self.notifyValue = string_to_dbus_byte_array(value)
+        # logger.debug(f"converted {self.notifyValue}")
```

Comment:
Do we need all this commented out code?

Reply:
not really, as the code is untested. wanted to keep it for debugging.

Reply:
np

Reply:
removed now.